test/functional/unified-spec-runner/operations.ts

```diff
@@ -173,11 +173,33 @@ operations.set('assertSessionTransactionState', async ({ entities, operation })
   expect(session.transaction.state).to.equal(driverTransactionStateName);
 });
 
+operations.set('assertNumberConnectionsCheckedOut', async ({ entities, operation }) => {
+  const client = entities.getEntity('client', operation.arguments.client);
+  const servers = Array.from(client.topology.s.servers.values());
+  const checkedOutConnections = servers.reduce((count, server) => {
+    const pool = server.s.pool;
+    return count + pool.currentCheckedOutCount;
+  }, 0);
+  process.nextTick(() => {
+    expect(checkedOutConnections).to.equal(operation.arguments.connections);
+  });
+});
+
 operations.set('bulkWrite', async ({ entities, operation }) => {
   const collection = entities.getEntity('collection', operation.object);
   return collection.bulkWrite(operation.arguments.requests);
 });
 
+operations.set('close', async ({ entities, operation }) => {
+  try {
+    const cursor = entities.getEntity('cursor', operation.object);
+    await cursor.close();
+  } catch (e) {
+    const changeStream = entities.getEntity('stream', operation.object);
```

Comment:
Why does this catch block attempt to close a change stream of the same name?

Reply:
The operation executes on both a change stream and a cursor, so it tries first to get a cursor from the entities, and if there is no cursor by that name an exception is thrown, so we catch it and attempt the same for a change stream of the same name.

Reply:
Maybe a suggestion for making this more clear would be to just turn this into existence checks?

```javascript
  const closable =
    entities.getEntity('cursor', operation.object, false) ??
    entities.getEntity('stream', operation.object, false);
  expect(closable).to.be.an('object');
  return closable.close();
```
Maybe something similar can be done for the `iterateUntilDocumentOrError` operation, it seems like this is a good use of skipping the assertExists flag in getEntity

Reply:
That's a nice suggestion but would you want the associated refactoring as part of this PR? The problem is that an entity does exist with the provided name, just the type of entity is different (`stream` vs `cursor`). So the exception thrown is not that the entity doesn't exist, rather the entity is the wrong type. So removing the existence checks still throws an exception in this case.

Reply:
🤦 Oh right, we can also just use .get to bypass the checking logic altogether. Or just leave behind a comment mentioning why we're doing this, if this were another language the catch block's error type might sufficiently hint at what's going on but here it's not clear what's being caught (kind of proven by my insufficient suggestion haha)

Reply:
Comment on reasoning added.